### PR TITLE
api: LEDS fixing field name

### PR DESCRIPTION
**Pull request by adisuissa**
Commit Message: api: LEDS fixing field name
Additional Description:
Previous name (`lb_endpoints_list`) was conflicting with the auto generated Java methods for the repeated `lb_endpoint` field. This change allows accessing the current `lb_endpoint` field using the same method names in the current java code.

Risk Level: Low - field is not being used.
Testing: N/A.
Docs Changes: N/A.
Release Notes: N/A.
Platform Specific Features: N/A.
Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md): Updating a not-implemented-hide field name.

Signed-off-by: Adi Suissa-Peleg <adip@google.com>

### CODE REVIEW

**Review by dmitri-d — DISMISSED** [review]

File: api/envoy/config/endpoint/v3/endpoint_components.proto [hunk 1]
```diff
@@ -141,15 +141,15 @@ message LocalityLbEndpoints {
...
+  // deprecated and replaced by *explicit_lb_endpoints_list*.]
   repeated LbEndpoint lb_endpoints = 2;
 
   // [#not-implemented-hide:]
   oneof lb_config {
     // The group of endpoints belonging to the locality.
     // [#comment:TODO(adisuissa): Once LEDS is implemented the *lb_endpoints* field
     // needs to be deprecated.]
-    LbEndpointList lb_endpoints_list = 7;
+    LbEndpointList explicit_lb_endpoints_list = 7;
```
Comment (mattklein123) [inline api/envoy/config/endpoint/v3/endpoint_components.proto:152]: Not to get into a huge bikeshedding situation, but we are going to live with this for a while and I'm not sure "explicit" has much meaning. Perhaps `load_balancer_endpoints` if we must name it something else? WDYT?

/wait-any

Comment (adisuissa) [reply api/envoy/config/endpoint/v3/endpoint_components.proto:152]: Sure, it's also a good name IMHO.
Let me update the PR.
/wait

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**